### PR TITLE
EE-1133

### PR DESCRIPTION
**Pull request by garethgooch**

### CODE REVIEW

File: app/services/SubmissionService.scala [hunk 1]
```diff
@@ -54,10 +54,16 @@ class SubmissionService @Inject()(
       subscriptionsToUpdate => {
         Future.sequence(subscriptionsToUpdate.map {
           case (year, psubs) =>
-            if (isDuplicateInSeqPsubs(psubs)) {Future.failed(SubmissionValidationException("Duplicate Psubs"))}
-            else {professionalBodiesService.validateYearInRange(psubs.map(_.name), year)}
+            if (isDuplicateInSeqPsubs(psubs)) {
```
Comment (amaalali) [inline app/services/SubmissionService.scala]: This seems a bit of an odd place to do this check, are there are any thoughts on how we can refactor to get this same goal.

File: app/services/ProfessionalBodiesService.scala [hunk 2]
```diff
@@ -18,41 +18,38 @@ package services
...
 
-    def professionalBodies(resourceLocation: String = frontendAppConfig.professionalBodiesList): Future[Seq[ProfessionalBody]] = {
-    environment.resourceAsStream(resourceLocation) match {
-      case Some(inputStream) =>
-        Json.parse(inputStream).validate[Seq[ProfessionalBody]] match {
-          case JsSuccess(value, _) => Future.successful(value)
-          case JsError(errors) => Future.failed(new Exception(s"failed to parse bodies: $errors"))
-        }
-      case _ => Future.failed(new Exception(s"failed to load bodies"))
+  private val resourceLocation: String = "professional-bodies.json"
```
Comment (amaalali) [inline app/services/ProfessionalBodiesService.scala]: Should we add the file name to the config file?

File: test/controllers/DuplicateClaimYearSelectionControllerSpec.scala [hunk 3]
```diff
@@ -282,7 +235,7 @@ class DuplicateClaimYearSelectionControllerSpec extends SpecBase with MockitoSug
 
     "redirect to Technical Difficulties for a POST when ProfessionalBodiesService fails" in {
 
-      when(mockProfessionalBodiesService.professionalBodies(any())).thenReturn(Future.failed(new Exception))
+      when(mockProfessionalBodiesService.professionalBodies).thenReturn(List.empty)
```
Comment (amaalali) [inline test/controllers/DuplicateClaimYearSelectionControllerSpec.scala]: I don't think we need this test, it's saying that the service successfully reads the file and but for some reason it's empty. We already have a test that checks that the list is non-empty so this scenario is invalid for the system

File: test/controllers/WhichSubscriptionControllerSpec.scala [hunk 4]
```diff
@@ -64,7 +59,7 @@ class WhichSubscriptionControllerSpec extends SpecBase with MockitoSugar with Sc
 
     "return OK and the correct view for a GET" in {
 
-      when(mockProfessionalBodiesService.professionalBodies()).thenReturn(Future.successful(Seq(ProfessionalBody("subscription", List(""),None))))
+      when(mockProfessionalBodiesService.professionalBodies).thenReturn(List(ProfessionalBody("subscription", List(""),None)))
```
Comment (amaalali) [inline test/controllers/WhichSubscriptionControllerSpec.scala]: Can we change these to be an empty list for the synonyms? Having an empty string could be misleading. If we think this is worth the change, there are a few more knocking about the test files that we should change too

```suggestion
      when(mockProfessionalBodiesService.professionalBodies).thenReturn(List(ProfessionalBody("subscription", Nil, None)))
```

File: test/services/ProfessionalBodiesServiceSpec.scala [hunk 5]
```diff
@@ -29,72 +29,30 @@ class ProfessionalBodiesServiceSpec extends SpecBase with MockitoSugar with Scal
 
   private val professionalBodiesService = new ProfessionalBodiesService(Environment.simple(), frontendAppConfig)
 
+  // TODO: Move this into an integration test suite
   "ProfessionalBodiesService" must {
     "professionalBodies" when {
       "must return a sequence of professional bodies" in {
-        val result = professionalBodiesService.professionalBodies()
+        val x = professionalBodiesService.professionalBodies
```
Comment (amaalali) [inline test/services/ProfessionalBodiesServiceSpec.scala]: We don't need this here
```suggestion
```

File: test/services/ProfessionalBodiesServiceSpec.scala [hunk 6]
```diff
@@ -29,72 +29,30 @@ class ProfessionalBodiesServiceSpec extends SpecBase with MockitoSugar with Scal
...
-
-      "no file must thrown an exception as Stream fails" in {
-        val result = professionalBodiesService.professionalBodies("no-file.json")
-
-        whenReady(result.failed) {
-          result =>
-            result mustBe an[Exception]
-            result.getMessage must include("failed to load bodies")
-        }
+        x.head mustEqual ProfessionalBody("100 Women in Finance Association", List.empty, Some(2018))
```
Comment (amaalali) [inline test/services/ProfessionalBodiesServiceSpec.scala]: We can just do 
```suggestion
        professionalBodiesService.professionalBodies.head mustEqual ProfessionalBody("100 Women in Finance Association", List.empty, Some(2018))
```

File: test/services/ProfessionalBodiesServiceSpec.scala [hunk 7]
```diff
@@ -29,72 +29,30 @@ class ProfessionalBodiesServiceSpec extends SpecBase with MockitoSugar with Scal
 
   private val professionalBodiesService = new ProfessionalBodiesService(Environment.simple(), frontendAppConfig)
 
+  // TODO: Move this into an integration test suite
```
Comment (amaalali) [inline test/services/ProfessionalBodiesServiceSpec.scala]: Do we think that this is worth doing? It does interact with the file system

File: app/services/ProfessionalBodiesService.scala [hunk 8]
```diff
@@ -18,41 +18,38 @@ package services
...
-        psubNames.forall {
-          name =>
-            allBodies.filter(_.name == name).map {
-              pBody => pBody.startYear.forall(_ <= year)
-            }.headOption.getOrElse(true)
-        }
-    }.map {
-      case true => true
-      case false => throw SubmissionValidationException("Year out of range")
+  def validateYearInRange(psubName: String, year: Int): Boolean = {
```
Comment (amaalali) [inline app/services/ProfessionalBodiesService.scala]: We need to add a test for this new method

File: app/services/ProfessionalBodiesService.scala (hunk 2 above)
Comment (amaalali) [reply app/services/ProfessionalBodiesService.scala]: The team decided to not move to config

File: test/services/ProfessionalBodiesServiceSpec.scala (hunk 7 above)
Comment (amaalali) [inline test/services/ProfessionalBodiesServiceSpec.scala]:
```suggestion
```

Comment (amaalali) [reply test/services/ProfessionalBodiesServiceSpec.scala]: We've decided to not move this into a separate test suite since it is a core part of the system
